internal/provisioner/utility/argo_utility.go

```diff
@@ -0,0 +1,190 @@
+		if err != nil {
+			return errors.Wrap(err, "failed to substitute values")
+		} else {
+			logger.WithField("Check the output file:", outputFilePath).Info("Substitution successful.")
+		}
+	}
+
+	commitMsg := "Adding: utility:" + utilityName + " to cluster: " + clusterID
+	if err = gitClient.Commit(tempDir+"/apps", commitMsg, logger); err != nil {
+		return errors.Wrap(err, "failed to commit to repo")
+	}
+
+	if err = gitClient.Push(logger); err != nil {
+		return errors.Wrap(err, "failed to push to repo")
+	}
+
+	appName := utilityName + "-sre-" + awsClient.GetCloudEnvironmentName() + "-" + clusterID
+	gitopsAppName := "gitops-sre-" + awsClient.GetCloudEnvironmentName()
+	app, err := argocdClient.SyncApplication(gitopsAppName)
+	if err != nil {
+		return errors.Wrap(err, "failed to sync application")
+	}
+
+	var wg sync.WaitGroup
+	timeout := time.Second * 300
+
+	wg.Add(1)
+	go argocdClient.WaitForAppHealthy(appName, &wg, timeout)
+	wg.Wait()
+
+	logger.WithField("app:", app.Name).Info("Deployed utility successfully.")
+
+	return nil
+}
+
+func (group utilityGroup) RemoveUtilityFromArgocd() error {
+
+	appsFile, err := os.ReadFile(group.tempDir + "/apps/" + group.awsClient.GetCloudEnvironmentName() + ArgocdAppsFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to read cluster file")
+	}
+	argoAppFile, err := argocd.ReadArgoApplicationFile(appsFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to read argo application file")
+	}
+
+	for _, utility := range group.utilities {
+		argocd.RemoveClusterIDLabel(argoAppFile, utility.Name(), group.cluster.ID, group.logger)
+
+		var modifiedYAML []byte
+		modifiedYAML, err = yaml.Marshal(argoAppFile)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal argo application file")
+		}
+		if err = os.WriteFile(group.tempDir+"/apps/"+group.awsClient.GetCloudEnvironmentName()+ArgocdAppsFile, modifiedYAML, 0644); err != nil {
+			return errors.Wrap(err, "failed to write argo application file")
+		}
+	}
+
+	if err = os.RemoveAll(group.tempDir + "/apps/" + group.awsClient.GetCloudEnvironmentName() + "/helm-values/" + group.cluster.ID); err != nil {
+		return errors.Wrap(err, "failed to remove helm values directory")
+	}
+
+	return nil
+
+}
+
+func substituteValues(inputFilePath, outputFilePath string, replacements map[string]string) error {
+	inputFile, err := os.Open(inputFilePath)
+	if err != nil {
+		return err
+	}
+	defer inputFile.Close()
+
+	outputFile, err := os.Create(outputFilePath)
+	if err != nil {
+		return err
+	}
+	defer outputFile.Close()
+
+	scanner := bufio.NewScanner(inputFile)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		for placeholder, replacement := range replacements {
+			line = strings.ReplaceAll(line, placeholder, replacement)
+		}
+
+		_, err := fmt.Fprintln(outputFile, line)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
While this works.. Consider using the built in Go template engine

Reply:
Makes sense, I'm going to open a task for it. Thanks